src/main/java/se/sundsvall/parkingpermit/integration/casedata/mapper/CaseDataMapper.java

```diff
@@ -31,27 +34,42 @@ public class CaseDataMapper {
 
 	private CaseDataMapper() {}
 
-	public static PatchErrand toPatchErrand(String externalCaseId, String phase, String phaseStatus, String phaseAction, String displayPhase) {
-		final var patchErrand =  toPatchErrand(externalCaseId, phase, phaseStatus, phaseAction);
-		return displayPhase != null ? patchErrand.addExtraParametersItem(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE).addValuesItem(displayPhase)) :
-			patchErrand.addExtraParametersItem(new ExtraParameter(CASEDATA_KEY_DISPLAY_PHASE));
+	public static PatchErrand toPatchErrand(final String externalCaseId, final String phase, final String phaseStatus, final String phaseAction, final String displayPhase, final List<ExtraParameter> extraParameters) {
+		final var patchErrand =  toPatchErrand(externalCaseId, phase, phaseStatus, phaseAction, extraParameters);
+
+		var filteredList = patchErrand.getExtraParameters().stream()
+			.filter(extraParameter -> !extraParameter.getKey().equals(CASEDATA_KEY_DISPLAY_PHASE))
+			.toList();
```

Comment:
.collect(Collectors.toCollection(ArrayList::new) result in a list that is not imutable. No need for creation of two lists.